modules/nw-sriov-example-dpdk-line-rate.adoc

```diff
@@ -0,0 +1,278 @@
+[source,yaml]
+----
+    ---
+    apiVersion: v1
+    kind: Namespace
+    metadata:
+      name: dpdk-test
+    ---
+    apiVersion: v1
+    kind: Pod
+    metadata:
+     annotations:
+       k8s.v1.cni.cncf.io/networks: '[ <1>
+         {
+          "name": "dpdk-network-1",
+          "namespace": "dpdk-test"
+         },
+         {
+          "name": "dpdk-network-2",
+          "namespace": "dpdk-test"
+         }
+       ]'
+       irq-load-balancing.crio.io: "disable" <2>
+       cpu-load-balancing.crio.io: "disable"
+       cpu-quota.crio.io: "disable"
+     labels:
+       app: dpdk
+     name: testpmd
+     namespace: dpdk-test
+    spec:
+     runtimeClassName: performance-performance <3>
+     containers:
+       - command:
+           - /bin/bash
+           - -c
+           - sleep INF
+         image: registry.redhat.io/openshift4/dpdk-base-rhel8
+         imagePullPolicy: Always
+         name: dpdk
+         resources:
+           limits:
+             cpu: "16"
+             hugepages-1Gi: 8Gi
+             memory: 2Gi
+           requests:
+             cpu: "16"
+             hugepages-1Gi: 8Gi
+             memory: 2Gi
+         securityContext:
+           capabilities:
+             add:
+               - IPC_LOCK
+               - SYS_RESOURCE
+               - NET_RAW
+               - NET_ADMIN
+           runAsUser: 0
+         volumeMounts:
+           - mountPath: /mnt/huge
+             name: hugepages
+     terminationGracePeriodSeconds: 5
+     volumes:
+       - emptyDir:
+           medium: HugePages
+         name: hugepages
+----
+<1> Request the SR-IOV networks you need: resources for the devices will be injected automatically.
+<2> Disable the CPU and IRQ load balancing base. See "Disabling interrupt processing for individual pods" for more information.
+<3> Set the `runtimeClass` to `performance-performance`.
+[NOTE]
+====
+Do not use `HostNetwork` or `privileged`.
+====
+[NOTE]
+====
+Request an equal number of resources for requests and limits to start the pod with guaranteed QOS.
+====
+[NOTE]
+====
+Do not start the pod with `SLEEP` and then exec into the pod to start the testpmd or the DPDK workload. This can add additional interrupts as the `exec` process is not pinned to any CPU.
+====
+
+== Running `testpmd`
+
+The following is an example script for running `testpmd`:
+
+[source,yaml]
+----
+    set -ex
+    export CPU=$(cat /sys/fs/cgroup/cpuset/cpuset.cpus)
+    echo ${CPU}
+
+    dpdk-testpmd -l ${CPU} -a ${PCIDEVICE_OPENSHIFT_IO_DPDK_NIC_1} -a ${PCIDEVICE_OPENSHIFT_IO_DPDK_NIC_2} -n 4 -- -i --nb-cores=15 --rxd=4096 --txd=4096 --rxq=7 --txq=7 --forward-mode=mac --eth-peer=0,50:00:00:00:00:01 --eth-peer=1,50:00:00:00:00:02
+----
+In this example, we use two different `sriovNetwork` CRs. The environment variable contains the VF PCI address that was allocated for the pod. If you use the same network in the pod definition, you must split the `pciAddress`.
```

Comment:
The ISG recommends not using "we" to describe the authors and audience. You could update the first sentence as follows:

> This example uses two different `sriovNetwork` CRs.

Reply:
You could provide the full name before the first use of the "CR" and "VF" acronyms, if this becomes a separate section.

Reply:
Done.